<launch>
  <node pkg="rxtools" name="rxconsole" type="rxconsole"/>

  <!-- you should first setup your camera grabbing/rectification nodes,
       see image_pipeline documentation. -->
  <remap from="camera/image_raw" to="camera/image_rect/"></remap>
  <node pkg="usb_cam" name="camera" type="usb_cam_node" output="log">
		
		<param name="io_method" value="mmap"/><!-- only option that works-->
		<param name="pixel_format" value="mjpeg"/><!-- or yuyv but slower-->
		
		<param name="image_width" value="1280"/><param name="image_height" value="720"/> <!--max res 30FPS acheivable depending on light and encoding-->
		<param name="image_width" value="800"/><param name="image_height" value="600"/> <!--max res 30FPS depending on light-->
		<param name="image_width" value="640"/><param name="image_height" value="480"/> <!-- 30fps acheivable depending on light-->
		
		<rosparam param="D">[0.032095, -0.238155, 0.000104, -0.002138, 0.0000]</rosparam>
        <rosparam param="K">[723.715912, 0.000000, 318.180121, 0.000000, 719.919423, 280.697379, 0.000000, 0.000000, 1.000000]</rosparam>
        <rosparam param="R">[1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0]</rosparam>
        <rosparam param="P">[723.715912, 0.000000, 318.180121, 0.000000, 0.000000, 719.919423, 280.697379, 0.000000, 0.000000, 0.000000, 1.000000, 0.000000]</rosparam>
	
		<!-- doesn't do anything noticable <param name="autofocus" value="true"/> -->
  </node>
  
  <!-- tracking node -->
  <node pkg="visp_tracker" type="tracker" name="tracker_mbt">
    <param name="camera_prefix" value="camera" />
  </node>

  <!-- starts the client while setting the camera input and the model
       which will be tracked -->
  <node pkg="visp_tracker" type="client" name="tracker_mbt_client" output="screen">
    <param name="camera_prefix" value="camera" />
    <param name="model_name" value="laas-box" />
    <param name="model_path" value="file:///home/tom/dev/usb_vision_servo/models" />
  </node>
</launch>
